### PR TITLE
ptfs: refine implementation of seal_size_check()

### PR DESCRIPTION
**Pull request by jiangliu**
Refine implementation of seal_size_check() with better handling of fallocate flags.

### CODE REVIEW

File: src/passthrough/mod.rs [hunk 1]
```diff
@@ -813,30 +813,36 @@ impl<S: BitmapSlice + Send + Sync> PassthroughFs<S> {
...
+                }
             }
 
             // setattr operation should be handled in setattr handler, other operations won't
             // change file size.
-            _ => Ok(()),
+            _ => return Err(enosys()),
         }
+
+        Ok(())
```
Comment (eryugey) [inline src/passthrough/mod.rs:845]: I don't quite get this hunk of change, other operations won't change file size, why return enosys here? And the last Ok(()) seems not reachable and can be removed?

Comment (jiangliu) [reply src/passthrough/mod.rs:845]: We have explicitly handled all operations defined by fallocate, `_` catches any new operation code, so it's `ENOSYS`.
`WRITE` will fall to the last `Ok(())` when it's legal.

Comment (jiangliu) [reply src/passthrough/mod.rs:844]: Refined the code a bit, maybe we should support `SETATTR` too.

File: src/passthrough/mod.rs [hunk 2]
```diff
@@ -813,30 +813,35 @@ impl<S: BitmapSlice + Send + Sync> PassthroughFs<S> {
...
+                        if size + offset > file_size {
+                            return Err(eperm());
+                        }
+                    }
+                    // collapse and insert will change file size, forbid.
+                    libc::FALLOC_FL_COLLAPSE_RANGE | libc::FALLOC_FL_INSERT_RANGE => {
+                        return Err(eperm());
+                    }
+                    // Invalid operation
+                    _ => return Err(enosys()),
```
Comment (eryugey) [inline src/passthrough/mod.rs:836]: Return EINVAL on invalid flags?

Comment (jiangliu) [reply src/passthrough/mod.rs]: Fixed

**Review by eryugey — APPROVED** [review]

**Pull request merged**